miqa/core/rest/accounts.py

```diff
@@ -0,0 +1,84 @@
+from django.conf import settings
+from django.contrib.auth import logout
+from django.contrib.auth.models import User
+from django.core.mail import EmailMultiAlternatives
+from django.shortcuts import redirect, render
+from django.views.generic.base import TemplateView
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+
+class AccountInactiveView(TemplateView):
+    template_name = 'account_inactive.html'
+
+
+class AccountActivateView(TemplateView):
+    def dispatch(self, request, *args, **kwargs):
+        if not request.user.is_authenticated and request.user.is_superuser:
+            return redirect('home')
```

Comment:
This isn't working in my production deployment. I was somehow able to approve myself without logging in at all.
```suggestion
        if not (request.user.is_authenticated and request.user.is_superuser):
            return redirect('home')
```
Perhaps python was parsing it as `(not authenticated) and superuser: go(home)`